ui/app/pages/confirm-token-transaction-base/confirm-token-transaction-base.container.js

```diff
@@ -44,7 +44,7 @@ const mapStateToProps = (state, ownProps) => {
   const tokenData = getTokenData(data)
   const tokenValue = getTokenValueParam(tokenData)
   const toAddress = getTokenAddressParam(tokenData)
-  const tokenAmount = tokenData && calcTokenAmount(tokenValue, decimals).toNumber()
+  const tokenAmount = tokenData && calcTokenAmount(tokenValue, decimals).toString()
```

Comment:
We should be explicitly declaring a base here, to prevent exponentiation. Apparently `bignumber.js` will use base 10 by default  with exponential notation for large and small values, but will use base 10 without exponential notation if it's specified explicitly.

The same goes for the other places where `.toString` was called on the return value of `calcTokenAmount` in this PR